### PR TITLE
Look for module of the definition in RDT during rename

### LINKED ISSUE #979: Library function is renamable
Open `proj` in VS Code. The values from `lib` should be resolved.

Rename `make_foo` in main.py succeeds, but should not.

[librename.zip](https://github.com/Microsoft/python-language-server/files/3091225/librename.zip)

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Fixes #979 

Variables do not have declaring module so we need to look in RDT, like goto definition does.

Includes https://github.com/microsoft/python-language-server/pull/1788 by accident so it needs to go last. Actual change is one file at the end.

### CODE REVIEW

**Comment by jakebailey** [comment]
#1788 is merged, so this needs a master merge to fix the diff.

File: src/LanguageServer/Impl/Sources/ReferenceSource.cs [hunk 1]
```diff
@@ -54,13 +54,19 @@ internal sealed class ReferenceSource {
...
 
                 var rootDefinition = definingMember.GetRootDefinition();
-                var name = definingMember.GetName();
-
-                // If it is an implicitly declared variable, such as function or a class
-                // then the location is invalid and the module is null. Use current module.
-                var declaringModule = rootDefinition?.DeclaringModule ?? analysis.Document;
-                if (!string.IsNullOrEmpty(name) && (declaringModule.ModuleType == ModuleType.User || options == ReferenceSearchOptions.All)) {
-                    return await FindAllReferencesAsync(name, declaringModule, rootDefinition, location, definitionSource, cancellationToken);
+                if (rootDefinition != null) {
```
Comment (jakebailey) [inline src/LanguageServer/Impl/Sources/ReferenceSource.cs]: This is an overall style nit, but this function looks pretty nested even though it's just a few null checks deep that could probably all return early. `url == null`, `definition == null`, `moduleUri == null`, etc.

**Comment by jakebailey** [comment]
Is there any way to test this fix?

**Comment by MikhailArkhipov** [comment]
Probably yes, need to add file to what we have in tests so it is considered a library. Let me try that.

**Pull request merged**